### PR TITLE
7 post api articles articleid comments

### PR DESCRIPTION
**Pull request by DexterGundar**
tasks 7 POST /api/articles/:article_id/comments and 8 PATCH /api/articles/:article_id finished

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -116,3 +116,161 @@ describe('get articles',()=>{
...
 })
+
+describe('POST /api/articles/:article_id/comments',()=>{
+  test("return 201 and newly posted comment", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: 'this is my sensible and nice comment'
+    }
+    return request(app)
+      .post("/api/articles/6/comments")
```
Comment (pippaamy) [inline __tests__/app.test.js:127]: great testing, one more for post could be a 201 that ignores unnecessary properties

File: __tests__/app.test.js [hunk 2]
```diff
@@ -116,3 +116,161 @@ describe('get articles',()=>{
...
+  .then(({body}) => {
+      expect(body.message).toBe('Bad Request');
+  })
+  });
+
+  test('return 400 and message if votes are not a number', () => {
+    const newVotes = {
+        inc_votes: '7asd'
+    }
+    return request(app).patch("/api/articles/3")
```
Comment (pippaamy) [inline __tests__/app.test.js:258]: great testing

File: __tests__/app.test.js [hunk 3]
```diff
@@ -116,3 +116,161 @@ describe('get articles',()=>{
...
+    .send(newVotes)
+    .expect(400)
+    .then(({body}) => {
+        expect(body.message).toBe('Voting must contain only numbers');
+    })
+    });
+    test('return 400 and message if wrong field is accessed', () => {
+      const newVotes = {
```
Comment (pippaamy) [inline __tests__/app.test.js:266]: this could be a 200, and be an ignored property

File: controllers/app.controllers.js [hunk 4]
```diff
@@ -34,4 +34,44 @@ exports.getArticleById = (req, res, next) => {
...
+
+exports.postComment = (req, res, next) =>{
+    const {article_id} = req.params
+    const newComment = req.body
+
+    if (Object.keys(newComment).length < 2 ||
+    !newComment.username ||
+    !newComment.body ||
+    article_id < 1) {
+    return res.status(400).send({ message: "Invalid data sent" });
```
Comment (pippaamy) [inline controllers/app.controllers.js:47]: we want error messages in the model rather than the controller

File: controllers/app.controllers.js [hunk 5]
```diff
@@ -34,4 +34,44 @@ exports.getArticleById = (req, res, next) => {
...
+  }
+
+    if (isNaN(article_id)) return next({ status: 400, message: 'Not a number, please enter valid id'});
```
Comment (pippaamy) [inline controllers/app.controllers.js:50]: here too

**Review by pippaamy — COMMENTED** [review]
good work, some changes to make and then feel free to merge!

File: models/app.models.js [hunk 6]
```diff
@@ -6,12 +6,14 @@ exports.fetchTopics = () => {
     })
 }
 
-exports.fetchArticleById = (id) =>{
+exports.fetchArticleById = (article_id) =>{
+  if (isNaN(article_id)) return Promise.reject({ status: 400, message: 'Not a number, please enter valid id'});
```
Comment (NRMSMITH) [inline models/app.models.js:10]: This could be left to a PSQL error handler - as the query will reject if the article_id isn't an integer :)

**Review by NRMSMITH — APPROVED** [review]
Please ensure that you've addressed my comment before merging - no need for another PR :)

File: models/app.models.js (hunk 6 above)
Comment (DexterGundar) [reply models/app.models.js:10]: I fixed this now.

**Pull request merged**